ansible/eda-server
# [AAP-10505][RBAC] RBAC Integration with existing API endpoints

## Testing instructions

1. Ensure latest database migrations are applied: 
    ```shell
    task manage -- migrate
    ```
2. Run `create_initial_data` management command to populate database with pre-defined user roles:
    ```shell
    task manage -- create_initial_data
    ```
3. Create two regular users:
    ```shell
    task manage -- shell

    >>> from aap_eda.core.models import User
    >>> User.objects.create_user(username="alice", password="secret")
    <User: alice>
    >>> User.objects.create_user(username="bob", password="secret")
    <User: bob>
    >>> User.objects.create_user(username="carol", password="secret")
    <User: carol>

    Do you really want to exit ([y]/n)? y
     ```

4. Assign roles to the created users:
    ```shell
    task manage -- add_role_to_user alice Viewer
    task manage -- add_role_to_user bob Admin
    ```
    User `alice` has read only permissions. User `bob` has full permissions. User `carol` has no permissions.

5. Test API with different user credentials:
    ```
    http -a alice:secret http://localhost:8000/api/eda/v1/projects/
    http -a carol:secret http://localhost:8000/api/eda/v1/projects/
    ```
